### PR TITLE
TypeScript

### PR DESCRIPTION
**Pull request by zacanger**
## Change Type

* [x] Feature
* [ ] Chore
* [ ] Bug Fix

## Change Level

* [x] major
* [ ] minor
* [ ] patch

### CODE REVIEW

**Comment by TomPridham** [comment]
i paused the code climate bot until youre done

**Comment by zacanger** [comment]
Marking a check not required doesn't pause the check, it just means it won't block the merge. I added test files to the ignore regex in CodeClimate and marked some things `wontfix`.

This is just about done, just trying to get styleguidist to work again.

**Comment by TomPridham** [comment]
doesn't this pause it? 
Screen Shot 2019-05-01 at 11 47 29 AM

**Comment by zacanger** [comment]
Oh yeah, that does.

File: .styleguide/section-heading.tsx [hunk 1]
```diff
@@ -1,21 +1,24 @@
-// @flow
-
-/* eslint-disable flowtype/no-weak-types */
-
 import * as React from 'react'
 import cx from 'classnames'
-import Heading from 'react-styleguidist/lib/rsg-components/Heading/index'
-import Styled from 'react-styleguidist/lib/rsg-components/Styled/index'
-import { colors } from '../src/theme.js'
+import Heading from 'react-styleguidist/lib/client/rsg-components/Heading/index'
```
Comment (zacanger) [inline .styleguide/section-heading.tsx]: These paths changed in styleguidist 9

File: src/utils/create-breakpoints/create-breakpoints.md [hunk 2]
```diff
@@ -17,4 +17,4 @@ You can access the following from breakpoints:
 - only: `(Key) => string`
   - returns a media query string for a breakpoint only at that value,
 - width: `(Key) => string`
-  - utility function to access a value at a given key,
\ No newline at end of file
+  - utility function to access a value at a given key,
```
Comment (zacanger) [inline src/utils/create-breakpoints/create-breakpoints.md]: newline

File: src/utils/create-breakpoints/create-breakpoints.ts [hunk 3]
```diff
@@ -53,12 +51,12 @@ const between = (start: Breakpoint, end: Breakpoint): string => {
 const only = (key: Breakpoint): string => between(key, key)
 const width = (key: Breakpoint): number => values[key]
 
-export default {
-  keys,
-  values,
-  up,
-  down,
+export const createBreakpoints = {
```
Comment (zacanger) [inline src/utils/create-breakpoints/create-breakpoints.ts]: Changed this to a named export

File: src/utils/from-theme/from-theme.md [hunk 4]
```diff
@@ -1,4 +1,4 @@
 `import { fromTheme } from 'jane-components'` 
 
 Pull values off the `theme` you've passed to the Styled Components
-`ThemeProvider`. Example (in a Styled Component): `background-color: ${fromTheme('bg')};`
\ No newline at end of file
+`ThemeProvider`. Example (in a Styled Component): `background-color: ${fromTheme('bg')};`
```
Comment (zacanger) [inline src/utils/from-theme/from-theme.md]: newline

File: package.json [hunk 5]
```diff
@@ -12,24 +12,21 @@
   ],
   "scripts": {
     "build": "run-p -s build:*",
-    "build:babel": "babel --ignore \"src/**/*.test.js\" src/ -d lib/",
+    "build:babel": "babel --extensions .ts,.tsx --ignore \"src/**/*.test.tsx\" src/ -d lib/",
     "build:rollup": "rollup -c",
-    "copy-flow": "flow-copy-source src lib",
```
Comment (TomPridham) [inline package.json]: do we need a step like this that copies `.d.ts` files over?

File: src/button/button.tsx [hunk 6]
```diff
@@ -1,37 +1,35 @@
...
+  fullWidth?: boolean
   /**
    * Use this prop if your button's only child is an icon
    */
-  iconOnlyButton?: boolean,
+  iconOnlyButton?: boolean
 }
 
-const Button = ({ testID, ...props }: ButtonProps): React$Element<*> => (
+const Button = ({ testID, ...props }: ButtonProps): React.ReactElement => (
```
Comment (TomPridham) [inline src/button/button.tsx]: is `React.Element` the recommended TS way?

Comment (zacanger) [reply src/button/button.tsx]: Not sure. It works, though.

**Pull request merged**